python/mxnet/optimizer.py

```diff
@@ -342,17 +342,20 @@ def create_state(self, index, weight):
         momentum = None
         weight_master_copy = None
         if self.multi_precision and weight.dtype == numpy.float16:
+            assert(weight.stype == 'default'), \
```

Comment:
I added an assert here because it's not yet tested. Let me add a test and remove this.